dataprofiler/data_readers/data_utils.py

```diff
@@ -99,7 +115,11 @@ def unicode_to_str(data, ignore_dicts=False):
     return data
 
 
-def json_to_dataframe(json_lines, selected_columns=None, read_in_string=False):
+def json_to_dataframe(
+    json_lines: List[Dict],
```

Comment:
looking deeper into this, I believe this might be incorrect.

` json_lines: List[Dict]` -> json lines can be a `List[Union[Dict, List, str]` Need to validate if it can be a number, but it might be able to also be a `float` `int` as well.

Reply:
This might be causing the issues below.

Reply:
https://docs.python.org/3/library/json.html#json-to-py-table
Could it be any of these? Or are there constraints on what kind of data we support?

Reply:
That's a fair question, I actually believe after looking through the code that it could be a list of any of those. We can use an alias to describe this maybe?
`JSON = t.Union[str, int, float, bool, None]` I'm not including the recursive option for now.
https://github.com/python/typing/issues/182

Reply:
Okay, after further review, while this can accept files which are lists or dicts

Reply:
that's bc:
```python
    elif first_item_type == dict:
        df = pd.json_normalize(json_lines)
    else:
        df = pd.DataFrame(json_lines)
```

Reply:
I was able to resolve it in https://github.com/capitalone/DataProfiler/pull/662/commits/7d5b8fb4f120504a8d84ea0da8a907d66798711b using 
```py
JSONType = Union[str, int, float, bool, None, List['JSONType'], Dict[str, 'JSONType']]
```
Note that this uses recursive types which are new to mypy. Currently you have to use the flag `--enable-recursive-aliases` but according to https://github.com/python/mypy/issues/731#issuecomment-1213482527 it will soon be a standard feature.

Reply:
Can we choose not to include recursive aliases? I think I mentioned above leaving out the `List['JSONType'], Dict[str, 'JSONType']` for now. Can we just include `Dict` in there instead?

Reply:
I'm concerned about conditionals being used @taylorfturner

Reply:
Changed to be nonrecursive.